packages/next/src/server/app-render.tsx

```diff
@@ -1744,10 +1755,24 @@ export async function renderToHTMLOrFlight(
           integrity: subresourceIntegrityManifest?.[polyfill],
         }))
 
+      const metadata = elementsFromResolvedMetadata(
+        await resolveMetadata(metadataItems, (filepath: string) => {
+          if (!serverComponentManifest) {
```

Comment:
```suggestion
          // Since we collected all the metadata items in a server loader,
          // it's possible that a module is a client module which has `.mod`
          // in the module reference representation. So we need the same
          // logic as RSC & SSR to resolve the actual module in the client
          // layer.
          if (!serverComponentManifest) {
```